Tests/Algorithm/AlgorithmAddDataTests.cs

```diff
@@ -228,8 +227,14 @@ private class TestHistoryProvider : IHistoryProvider
             public string underlyingSymbol2 = "AAPL";
             public int DataPointCount { get; }
             public Resolution LastResolutionRequest;
-            public void Initialize(AlgorithmNodePacket job, IDataProvider dataProvider, IDataCacheProvider dataCacheProvider,
-                IMapFileProvider mapFileProvider, IFactorFileProvider factorFileProvider, Action<int> statusUpdate)
+
+#pragma warning disable CS0067 // The event is never used
```

Comment:
Regarding these not being used, my earlier statement was a misunderstanding. They're not used because the class doesn't call `event.Invoke` as I'm sure you figured out -- I'm catching up to you :)
I think renaming these events to be more 'eventy' will resolve one concern about this issue, IOW, it's OK if a certain implementation doesn't raise a certain event because it's not applicable.